src/sentry/deletions/base.py

```diff
@@ -213,7 +215,7 @@ def chunk(self) -> bool:
         query_limit = self.query_limit
         remaining = self.chunk_size
 
-        while remaining > 0:
+        while remaining >= 0:
```

Comment:
This is likely the fix. When we had 100 groups, we would say we have more groups to delete even though there's non-left.